client/landing/gutenboarding/components/domain-picker/index.tsx

```diff
@@ -93,87 +104,95 @@ const DomainPicker: FunctionComponent< Props > = ( {
 		[ search, queryParameters ]
 	);
 
-	let numberOfFreeDomains = 0;
-	let numberOfPaidDomains = 0;
-	const suggestions = allSuggestions?.filter( suggestion => {
-		if (
-			suggestion.is_free &&
-			! numberOfFreeDomains &&
-			suggestion.domain_name !== currentDomain?.domain_name
-		) {
-			numberOfFreeDomains++;
-			return suggestion;
-		}
-		if ( ! suggestion.is_free && numberOfPaidDomains < 5 ) {
-			numberOfPaidDomains++;
-			return suggestion;
-		}
-	} );
-
-	const handleDomainClick = ( suggestion: DomainSuggestion ) => {
-		if ( suggestion.is_free ) {
-			onDomainSelect( suggestion );
-		} else {
-			onDomainPurchase( suggestion );
-		}
-	};
+	const freeSuggestions = allSuggestions?.filter( suggestion => suggestion.is_free );
+	const paidSuggestions = allSuggestions?.filter( suggestion => ! suggestion.is_free );
+
+	// Recommend paid domain with exact site title match with highest relevance score.
+	const recommendedSuggestion = paidSuggestions?.find( suggestion =>
+		suggestion.match_reasons.includes( 'exact-match' )
+	);
```

Comment:
Are we supposed to only have 1 recommended suggestion? It seems arbitrary to pick one over another if multiple domains are exact matches.

Isn't there a score or something we can use as well?

Reply:
The results returned from the API is sorted by relevance score. That's why there's no additional sorting being done here.

Knowing that it is already sorted by score, it is possible to just do `const recommendedSuggestion = paidSuggestion[0]`. However, the domain with the highest score is weird. Take the one in the screenshot for example, I 
 searched for `jamaal` and the API gave me `jam-aal.com` as the highest score, which is not an **exact match**.

That's why I prioritized **exact match domain** with highest score over just **domain** with highest score.

The drawback is that sometimes the API doesn't return exact match, so the recommended badge doesn't appear.

We should have a discussion around the algorithm that decides what should be the recommended domain(s).

Reply:
Sounds like we should recommend the exact-match or the highest score if there's no exact-match. Find to leave for a follow-up 👍